### PR TITLE
Limit proxy nodes to one.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/core.rs [hunk 1]
```diff
@@ -1137,15 +1141,21 @@ impl Core {
...
         }
 
-        if let Some(previous_name) = self.proxy_map.insert(peer_id, public_id.clone()) {
-            warn!("Adding bootstrap node to proxy map caused a prior ID to eject. Previous name: \
-                   {:?}",
-                  previous_name);
-            warn!("Dropping this peer {:?}", peer_id);
-            self.crust_service.disconnect(&peer_id);
-            let _ = self.proxy_map.remove(&peer_id);
+        if self.proxy_map.len() == 0 {
```
Comment (afck) [inline src/core.rs]: Or `self.proxy_map.is_empty()` - I think Clippy will demand that. :-)

File: src/core.rs [hunk 2]
```diff
@@ -369,6 +369,10 @@ impl Core {
                 //     true
                 // });
                 trace!(" -----------------------------------");
+
+                trace!(" -----------------------------------");
+                trace!("| Number of proxy nodes : {}", self.proxy_map.len());
+                trace!(" -----------------------------------");
```
Comment (afck) [inline src/core.rs]: I wouldn't add that, since we now make sure that it can be at most 1 anyway.

Comment (ghost) [inline src/core.rs]: It was more for ensuring it goes back to zero when we have group size routing nodes in the table, but fair point.

**Comment by ghost** [comment]
@afck ^^^ Done.

**Pull request merged**